packages/ingest/src/recursion.ts

```diff
@@ -72,18 +70,18 @@ async function recurOnDirectory(
  * @returns {Promise} the promise of an object that has the path is the key and
  *   list of files as values
  */
-export async function recursion(
-  fn: RecursionFunction,
+export async function recursion<T extends RecursionFile>(
+  fn: (path: string) => Promise<T[]>,
   configuredPath: string
-): Promise<RecursionFile[]> {
+): Promise<T[]> {
   const normalizedPath = path.normalize(configuredPath);
   const isAbsolutePath = path.isAbsolute(normalizedPath);
   try {
     const segments = normalizedPath
       .split('/') // split on divider
       .filter((segment) => segment.trim() !== ''); // filter out empty strings from split
     const startingPath = isAbsolutePath ? '/' : '.';
-    return await recurOnDirectory(fn, startingPath, segments, -1);
+    return recurOnDirectory<T>(fn, startingPath, segments, -1);
```

Comment:
`return await` is required for any thrown error from the recursion to be caught, isn't it?

Reply:
89df42190